src/app/core/admin/user-details/user-details.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { User } from '../../user/user';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { EntityMapperService } from '../../entity/entity-mapper.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+@Component({
+  selector: 'app-user-details',
+  templateUrl: './user-details.component.html',
+  styleUrls: ['./user-details.component.scss'],
+})
+export class UserDetailsComponent implements OnInit {
+  creating: boolean = true;
+  private readonly defaultPasswordValue: string = 'password';
+  private readonly defaultUsernameValue: string = 'username';
+
+  userForm: FormGroup = this.fb.group({
+    username: [{value: this.defaultUsernameValue, disabled: !this.creating}, Validators.required],
+    password: [this.defaultPasswordValue, Validators.required],
+    admin: [false],
+  });
+
+  constructor(
+    private dialogRef: MatDialogRef<UserDetailsComponent>,
+    @Inject(MAT_DIALOG_DATA) public user: User,
+    private fb: FormBuilder,
+    private entityMapper: EntityMapperService,
+    private snackBar: MatSnackBar,
+  ) { }
+
+  ngOnInit() {
+    this.creating = this.user === null;
+    if (! this.creating) {
+      this.userForm.patchValue({username: this.user.name});
+      // Username can't be edited after creation
+      this.userForm.get('username').disable();
+      this.userForm.patchValue({admin: this.user.isAdmin()});
+    }
+  }
+
+  saveUser() {
+    const {username, password, admin} = this.userForm.getRawValue();
+    if (this.creating) {
+      this.user = new User(username);
+    } else if (username !== this.user.name) {
+      // Can't be reached because changing username is not allowed
+      this.entityMapper.remove<User>(this.user);
+      this.user = new User(username);
+    }
+    this.user.name = username;
+    if (password !== this.defaultPasswordValue) {
+      this.user.setNewPassword(password);
+    }
+    this.user.admin = admin;
+    this.entityMapper.save<User>(this.user)
```

Comment:
Saving the `User` entity is not sufficient in our current architecture - we have an additional user document in CouchDB's "_users" database that is used for authorization of access to the server database. New users and passwords need to be synchronously done there as well (this is what made #17 so complicated).